src/testsets.jl

```diff
@@ -111,16 +111,31 @@ end
 #################################
 # Accessing and setting methods #
 #################################
+
 """
-    properties(ts::ReportingTestSet)
-    properties(ts::AbstractTestSet)
+    properties(ts::AbstractTestSet) -> Union{Vector{Pair{String, Any}}, Nothing}
 
-Get the properties dictionary of a `ReportingTestSet`, returns
-nothing for an `AbstractTestSet`. Can be extended for custom
-`TestSet`s, and must return either a `Dict` or `nothing`.
+Get the properties of a `ReportingTestSet`. Can be extended for custom testsets.
+Returns `nothing` for an `AbstractTestSet`.
+
+See also: [`recordproperty`](@ref) and [`recordproperty!`](@ref).
 """
+properties(::AbstractTestSet) = nothing
 properties(ts::ReportingTestSet) = ts.properties
-properties(ts::AbstractTestSet) = nothing
+
+"""
+    recordproperty!(ts::AbstractTestSet, name::AbstractString, value)
+
+Associate a property with the testset. Can be extended for custom testsets.
+
```

Comment:
```suggestion
Defaults to just discarding the information unless extended, e.g as is done in `ReportingTestSet`.
```

I am not sure if this is a good default.

Reply:
Maybe a compromise would be to define the no-op behavior for `Test.DefaultTestSet` and error for custom test sets.